### PR TITLE
[terra-popup] Update examples to use hooks

### LINKED ISSUE #925: [terra-popup] Code not shown for few examples in devSite.
## Description
In DevSite, example code is not shown for terra-popup in few examples.

## Steps to Reproduce
Navigate to https://engineering.cerner.com/terra-ui/components/terra-popup/popup/popup.
Implementation code is not shown for few examples.

## Expected Behavior
Implementation code should be shown.

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
Resolves #925

### CODE REVIEW

File: packages/terra-popup/src/terra-dev-site/doc/popup/Popup.1.doc.jsx [hunk 1]
```diff
@@ -34,18 +38,22 @@ const DocPage = () => (
       {
         title: 'Popup Display With Arrow',
         example: <PopupWithArrow />,
+        source: PopupWithArrowSrc,
```
Comment (lokesh-0813) [inline packages/terra-popup/src/terra-dev-site/doc/popup/Popup.1.doc.jsx]: `<PopupWithArrow />` needs to have detail for enabling arrow.

[this](https://github.com/cerner/terra-framework/blob/9fcea5284e7c4b7621b3675249f8d0dadd47e064/packages/terra-popup/src/terra-dev-site/doc/common/PopupCommonExample.jsx#L17) should default to `false` and other examples consuming PopupCommonExample should adjust accordingly.

Comment (pranav300) [reply packages/terra-popup/src/terra-dev-site/doc/popup/Popup.1.doc.jsx]: Fixed here https://github.com/cerner/terra-framework/pull/927/commits/1e41be63db4d72a46f315d694ebded26ff6f5b68

**Comment by dkasper-was-taken** [comment]
For the code examples of Attachments, Behaviors, and Size variants we deliberately do not want to display the src code to the user as the examples are created using a common reusable component with flexing variables and controls. As the result the code doesn't show the user anything meaningful and could complicate understanding.

We can however present clean examples for the default and the arrow examples. So if you want to create new examples for those two, it would improve the documentation.

File: packages/terra-popup/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ Changelog
 
 Unreleased
 ----------
+### Changed
+* Updated docs.
```
Comment (emilyrohrbough) [inline packages/terra-popup/CHANGELOG.md]: This should be a bit more descriptive

**Comment by pranav300** [comment]
I and @ShettyAkarsh had an offline conversation to convert all the examples to use hooks and display the code for each component.

File: packages/terra-popup/src/terra-dev-site/doc/example/PopupDimensions.jsx [hunk 3]
```diff
@@ -1,101 +1,97 @@
...
-            onRequestClose={this.handleRequestClose}
-            targetRef={() => document.getElementById('popup-dimensions')}
-          >
-            <PopupContent contentStyle={contentStyle} />
-          </Popup>
-          <Button id="popup-dimensions" text={`${this.state.popupContentHeight || 'Default'} x ${this.state.popupContentWidth || 'Default'} Popup`} onClick={this.handleButtonClick} />
-        </div>
+        <Popup
+          {...contentDimensions}
+          classNameArrow="test-arrow"
```
Comment (dkasper-was-taken) [inline packages/terra-popup/src/terra-dev-site/doc/example/PopupDimensions.jsx]: Shouldn't need classNameArrow or classNameContent in the example page.

File: packages/terra-popup/src/terra-dev-site/doc/example/PopupStandard.jsx [hunk 4]
```diff
@@ -1,6 +1,40 @@
...
+
+  const handleRequestClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <React.Fragment>
+      <Button text="Default Popup" onClick={handleButtonClick} refCallback={setButtonNode} />
+      <Popup
+        attachmentBehavior="auto"
```
Comment (dkasper-was-taken) [inline packages/terra-popup/src/terra-dev-site/doc/example/PopupStandard.jsx]: Shouldn't need attachmentBehavior or contentAttachment values set in the default example.

File: packages/terra-popup/src/terra-dev-site/doc/example/PopupDimensions.jsx (hunk 3 above)
Comment (pranav300) [reply packages/terra-popup/src/terra-dev-site/doc/example/PopupDimensions.jsx]: Fixed here https://github.com/cerner/terra-framework/pull/927/commits/f26bebb7c61ac4b5747385bd2a04d26a4785fb3d.

File: packages/terra-popup/src/terra-dev-site/doc/example/PopupStandard.jsx (hunk 4 above)
Comment (pranav300) [reply packages/terra-popup/src/terra-dev-site/doc/example/PopupStandard.jsx]: Fixed here https://github.com/cerner/terra-framework/pull/927/commits/f26bebb7c61ac4b5747385bd2a04d26a4785fb3d.

File: packages/terra-popup/src/terra-dev-site/doc/example/PopupAttachmentBehavior.jsx [hunk 5]
```diff
@@ -1,35 +1,63 @@
...
 import classNames from 'classnames/bind';
 import styles from './PopupAttachmentBehavior.module.scss';
 
 const cx = classNames.bind(styles);
 
-class PopupAttachmentBehavior extends React.Component {
-  constructor(props) {
-    super(props);
+function PopupAttachmentBehavior() {
+  const buttonElement = useRef(null);
```
Comment (jmsv6d) [inline packages/terra-popup/src/terra-dev-site/doc/example/PopupAttachmentBehavior.jsx]: You shouldn't need `null` here.

Comment (pranav300) [reply packages/terra-popup/src/terra-dev-site/doc/example/PopupAttachmentBehavior.jsx]: Fixed here https://github.com/cerner/terra-framework/pull/927/commits/6f013aa3d8244f7caf366c84fa97c41acdefcbcb.

**Pull request merged**